### ISSUE TITLE
Create version with wasm+wazero instead of cgo

### ISSUE THREAD

**Issue by redexp** [tree-sitter/go-tree-sitter#16]
Good day

If you don't know [wazero](https://github.com/tetratelabs/wazero) is fully go written WebAssembly runtime so with tree-sitter.wasm + wazero this project could be fully independent from cgo.

**Comment by amaanq** [comment]
This would be nice to add support for, and have it gated behind a build tag

**Comment by ObserverOfTime** [comment]
Why wazero and not [wasmtime](https://github.com/bytecodealliance/wasmtime-go)?

**Comment by redexp** [comment]
wazero don't have C dependency 

Also I thought maybe create separate repo with just latest wasm file so js project and any other can just import one same file

**Comment by amaanq** [comment]
It'd make more sense to use wasmtime though, since we already have the code in place for it in the C side

**Comment by topi314** [comment]
is there any specific reason this isn't planned anymore?

**Comment by redexp** [comment]
I think it's somehow related to [this issue](https://github.com/tree-sitter/tree-sitter/issues/4136)

Personally I decided to drop tree-sitter from my project and to write parser fully on Go. It's really easier to create parser from scratch than to compile tree-sitter with Go to wasm.

**Comment by amaanq** [comment]
Yeah, we're not getting rid of our C dependency, and if we're going to add wasm support, it'd be using wasmtime, not wazero
